src/authentication/UnsecureConstantCredentialsExtractor.ts

```diff
@@ -1,24 +1,25 @@
 import { getLoggerFor } from '../logging/LogUtil';
-import type { Credentials } from './Credentials';
+import type { Credentials, CredentialSet } from './Credentials';
 import { CredentialsExtractor } from './CredentialsExtractor';
+import { AGENT } from './CredentialTypes';
 
 /**
  * Credentials extractor that authenticates a constant agent
  * (useful for development or debugging purposes).
  */
 export class UnsecureConstantCredentialsExtractor extends CredentialsExtractor {
-  private readonly agent: Credentials;
+  private readonly credentials: CredentialSet;
```

Comment:
I originally started with a version where all the credential and permission stuff had generic typings. But that became a bit unmanageable since it was then also required for all the classes that took one as input and we couldn't guarantee the typings of the incoming data anyway.